src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/WorkspaceDiagnosticSourceHelpers.cs

```diff
@@ -47,7 +39,7 @@ internal abstract class AbstractWorkspaceDiagnosticSourceProvider(string name) :
         }
     }
 
-    protected static bool ShouldSkipDocument(RequestContext context, TextDocument document)
+    public static bool ShouldSkipDocument(this RequestContext context, TextDocument document)
```

Comment:
i would avoid having these be extensions.  Just call explicitly into the helper to make it clear that this is a helper, and not some instance functionality on RequestContext itself.